### ISSUE TITLE
Issue with sending a buffer

### ISSUE THREAD

**Issue by greenfork** [repl-electric/sonic-pi.el#20]
Hi! I had an issue with sending a buffer whereas sending a region works fine. Are there any caveats of sending a region of all the buffer instead of sending the buffer as a file? Is this a common problem?

And if it's ok, can I suggest a pull request of reimplementing the `sonic-pi-send-buffer` function in `sonic-pi-osc.el`?

**Comment by mbutz** [comment]
Can confirm that. 'sonic-pi-send-buffer' does not do anything anymore.

**Comment by greenfork** [comment]
@mbutz I'm currently using this replacement if you are interested
``` emacs lisp
(defun sonic-pi-send-full-buffer ()
  "send full buffer selecting region of full buffer"
  (interactive)
  (sonic-pi-osc-send-text (point-min) (point-max))
  (hlt-highlight-regexp-region (point-min)
                               (point-max)
                               ".+"
                               'eval-sonic-pi-flash nil)
  (run-at-time flash-time nil 'hlt-unhighlight-region nil nil nil))

(add-hook 'sonic-pi-mode-hook
          (lambda ()
            (setq sonic-pi-path "/opt/sonic-pi/app/")
            (define-key sonic-pi-mode-map (kbd "C-c C-p") 'sonic-pi-stop-all)
            (define-key sonic-pi-mode-map (kbd "C-c C-k") 'sonic-pi-send-full-buffer)))
```

**Comment by mbutz** [comment]
Wow. Thanks! Works flawlessly.

Just out of curiosity: Can you confirm #19 or haven't you tried to use midi yet?

**Comment by greenfork** [comment]
@mbutz I'm new to Sonic Pi and haven't played with the `midi` command yet. But yeah, I will post in #19 if I have something useful.

**Comment by josephwilk** [comment]
@greenfork Any chance you have the error in the buffer? Also the version of SP you are trying to run against?

Thanks.

**Comment by josephwilk** [comment]
I believe this is an issue that SP in latest versions removed the fn `save-and-run-buffer-via-local-file`.

https://github.com/repl-electric/sonic-pi.el/blob/master/sonic-pi-osc.el#L61-L66

This makes me sad, since this fn fixes a big problem with SP: #21

Probably I'll fix this by resorting back to the old SP osc fn. Since most people will just want it to work (until they hit #21 and cry).

**Comment by josephwilk** [comment]
Should be fixed now @greenfork, I'll close but feel free to re-open if I messed anything up.
